TrustKit/Pinning/pinning_utils.m

```diff
@@ -0,0 +1,90 @@
+/*
+
+ pinning_utils.m
+ TrustKit
+
+ Copyright 2023 The TrustKit Project Authors
+ Licensed under the MIT license, see associated LICENSE file for terms.
+ See AUTHORS file for the list of project authors.
+
+ */
+
+#import "pinning_utils.h"
+#include <dlfcn.h>
+#include "TargetConditionals.h"
+
+bool evaluateTrust(SecTrustRef serverTrust, SecTrustResultType *trustResult, NSError **error) {
+    bool isTrusted = false;
+
+    if (@available(iOS 12.0, macOS 14.0, tvOS 12.0, watchOS 5.0, *)) {
+        CFErrorRef errorRef;
+        isTrusted = SecTrustEvaluateWithError(serverTrust, &errorRef);
+        OSStatus status = SecTrustGetTrustResult(serverTrust, trustResult);
+        if (status != errSecSuccess)
+        {
+            isTrusted = false;
+            NSString *errDescription = [NSString stringWithFormat:@"got status %d", status];
+            if (error != NULL) {
+                *error = [[NSError alloc] initWithDomain:@"com.datatheorem.trustkit" code:1 userInfo:@{NSLocalizedDescriptionKey:errDescription}];
+            }
+        }
+        else if (!isTrusted && (error != NULL))
+        {
+            *error = (__bridge_transfer NSError *)errorRef;
+        }
+    }
+    else
+    {
+        // Use pragmas to supress deprecated warnings
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        OSStatus status = SecTrustEvaluate(serverTrust, trustResult);
+#pragma clang diagnostic pop
+        if (status == errSecSuccess) {
+            isTrusted = true;
```

Comment:
I don't think an `errSecSuccess` status is the same as the certificate being trusted. For example, the status can be `errSecSuccess` while `trustResult` is `fatalTrustFailure`, which indicates that the certificate is not trusted, I think.

I would suggest getting rid of `SecTrustEvaluate()` entirely, since `SecTrustEvaluateWithError()` is available starting with iOS 12?

Reply:
Good point. The naming of that variable is not appropriate for the return value of `SecTrustEvaluate()`.

Getting rid of `SecTrustEvaluate()` will simplify things but drop support for macOS 10.13 and watchOS 4. The newly supported platforms will be: 
```
iOS 12+, macOS 10.14+, tvOS 12+ or watchOS 5+ 
```
Should I go for it?

Reply:
Yes, let's do it. Getting rid of `SecTrustEvaluate()` (and its broken/dangerous API) will be a big improvement. Thanks!

Reply:
Done. Updated the readme with the new minimum OS versions